### ISSUE TITLE
Automate rebuilds with Travis CI

### ISSUE THREAD

**Issue by namurphy** [PlasmaPy/plasmapy.github.io#3]
It would probably be helpful if we could [automate Nikola rebuilds with Travis CI](https://getnikola.com/blog/automating-nikola-rebuilds-with-travis-ci.html).  I started to set this up, but unfortunately I was not able to get this to work.  If anyone with more (or any!) experience with Travis CI would like to take a look at this, please do!

**Comment by namurphy** [comment]
This is the error I got when I tried to trigger a rebuild:
```
0.01s$ openssl aes-256-cbc -K $encrypted_5efde3fde6ee_key -iv $encrypted_5efde3fde6ee_iv -in
id_rsa.enc -out id_rsa -d

iv undefined

The command "openssl aes-256-cbc -K $encrypted_5efde3fde6ee_key -iv $encrypted_5efde3fde6ee_iv -in id_rsa.enc -out id_rsa -d" failed and exited with 1 during .

Your build has been stopped.
```

Looks like a problem with RSA encryption.  I'll try to come back to this after I get some sleep. :smiley_cat:

**Comment by sinoroc** [comment]
(found you there: http://up-for-grabs.net)

This tutorial from Nikola seems unnecessary complicated. It might not have been the case when you started this effort, but Travis CI offers a much better way to solve this:
https://docs.travis-ci.com/user/deployment/pages/

Something similar to this should get you there:

```yaml
language: 'python'
install:
  - 'pip install --requirement requirements.txt'
script:
  - 'nikola build'
deploy:
  - provider: 'pages'
    skip_cleanup: true
    target-branch: 'master'
    on:
      branch: 'src'
    local-dir: 'output'
    github_token:
      secure: "..."
```

Have not actually tested this (might do later) and can't be certain it will work for sure, but hope this helps anyway.

**Comment by StanczakDominik** [comment]
Thanks for the heads up! I'll try to get this tested, but I'm already behind on a couple of other PRs so if you get the time to try it before I open up a PR for this, please do!

**Comment by sinoroc** [comment]
You asked about potential security issues with the GitHub token - it is fine since the secured version of the token is in the commit - and that led me to read again the chapter of the Travis CI documentation about environment variables. I believe now that we are not using the best solution here.

It would be much better to set the environment variable for the GitHub token from the Travis CI dashboard instead of within the `.travis.yml` file.
https://docs.travis-ci.com/user/environment-variables#Defining-Variables-in-Repository-Settings
This way each contributor can have its own value for the `GITHUB_TOKEN` variable.

I did not realize it was possible. I'll prepare a new pull request to improve this.

**Comment by StanczakDominik** [comment]
Closed by the mighty @sinoroc in #13 :)
